python/pyiceberg/catalog/rest.py

```diff
@@ -422,7 +424,6 @@ def list_namespaces(self, namespace: Union[str, Identifier] = ()) -> List[Identi
             ),
             headers=self.headers,
         )
-        response.raise_for_status()
         namespaces = ListNamespaceResponse(**response.json())
```

Comment:
We should move this one after the `raise_for_status` block, otherwise, we might consume the payload before checking if it is a 2xx. A non 2xx would have a different structure, and we would fail with a pedantic error.

Reply:
Good catch!